src/plugins/hotkeys/hotkeys.ts

```diff
@@ -0,0 +1,132 @@
+import type Player from 'core/player.js'
+
+type pluginParameter = {
+	player: Player
+	options: {
+		seekTime?: number
+		volumeStep?: number
+	}
+}
+
+/**
+ * Vlitejs Hotkeys plugin
+ * @module Vlitejs/plugins/Hotkeys
+ */
+export default class Hotkeys {
+	player: any
+	seekTime: number
+	volumeStep: number
+
+	providers = ['html5', 'youtube', 'dailymotion', 'vimeo']
+	types = ['video', 'audio']
+
+	/**
+	 * @constructor
+	 * @param options
+	 * @param options.player Player instance
+	 * @param options.options Plugins options
+	 */
+	constructor({ player, options = {} }: pluginParameter) {
+		this.player = player
+		this.seekTime = options.seekTime ?? 5
+		this.volumeStep = options.volumeStep ?? 0.1
+
+		this.onKeydown = this.onKeydown.bind(this)
+	}
+
+	/**
+	 * Initialize
+	 */
+	init() {
+		this.addEvents()		
+	}
+
+	/**
+	 * Add event listeners
+	 */
+	addEvents() {
+		this.player.container.addEventListener('keydown', this.onKeydown)
+	}
+
+	/**
+	 * On keydown event on the media element
+	 * @param e Event listener datas
+	 */
+	onKeydown(e: KeyboardEvent) {
+		const activeElement = document.activeElement
+		const { keyCode } = e
+
+		if (
+			[9, 32, 37, 39].includes(keyCode) &&
+			this.player.autoHideGranted &&
+			(activeElement === this.player.container || activeElement?.closest('.v-vlite'))
+		) {
+			// Stop and start the auto hide timer on selected key code
+			this.player.stopAutoHideTimer()
+			this.player.startAutoHideTimer()
+		} else if (
+			[37, 39].includes(keyCode) &&
+			(activeElement === this.player.container || activeElement === this.player.elements.progressBar)
+		) {
+			// Backward or forward video with arrow keys
+			e.preventDefault()	// Prevent default behavior on input range
+
+			if (keyCode === 37) {
+				this.fastForward('backward')
+			} else {
+				this.player.fastForward('forward')
+			}
+		} else if (
+			[38, 40].includes(keyCode) &&
+			(activeElement === this.player.container || activeElement === this.player.elements.volume)
+		) {
+			// Increase or decrease volume with arrow keys
+			if (keyCode === 38) {
+				this.increaseVolume()
+			} else {
+				this.decreaseVolume()
+			}
+		} else if (keyCode === 32 && activeElement === this.player.container) {
+			// Toggle the media playback with spacebar key
+			this.player.controlBar.togglePlayPause(e)
+		}
+	}
+
+	/**
+	 * Trigger the video fast forward (front and rear)
+	 * @param direction Direction (backward|forward)
+	 */
+	fastForward(direction: string) {
+		this.player.getCurrentTime().then((seconds: number) => {
+			this.player.seekTo(
+				direction === 'backward' ? seconds - this.seekTime : seconds + this.seekTime
```

Comment:
I prefer all plugin options accessible by `this.options` (linked to the thread about refactor from Cast plugin).

(Same for volume below)

Reply:
We can use directly the the class attributes to reduce weight and gain performance.

Related https://github.com/vlitejs/vlite/pull/165#discussion_r1976751229